### PR TITLE
[GO-105] add Context to workers

### PR DESCRIPTION
**Pull request by jenchik**

### CODE REVIEW

File: examples/workers/main.go [hunk 1]
```diff
@@ -35,8 +35,13 @@ func main() {
 	err := workers.New(workers.Options{
 		Name:     "worker #1",
 		Schedule: workers.Every(time.Second * 5),
-		Handler: func() {
-			fmt.Printf("[%s] worker #1 every 5 secs\n", time.Now().Format("15:04:05"))
+		Handler: func(c context.Context) {
+			select {
+			case <-c.Done():
+				fmt.Printf("[%s] worker #1; we stopping\n", time.Now().Format("15:04:05"))
```
Comment (im-kulikov) [inline examples/workers/main.go:41]: srsly?

File: examples/workers/main.go [hunk 2]
```diff
@@ -73,7 +78,7 @@ func main() {
 	err = workers.New(workers.Options{
 		Name:     "worker #3",
 		Schedule: sched,
-		Handler: func() {
+		Handler: func(context.Context) {
 			fmt.Printf("[%s] worker #3 every minute at 12 secs\n", time.Now().Format("15:04:05"))
```
Comment (im-kulikov) [inline examples/workers/main.go:82]: srsly?

File: examples/workers/main.go [hunk 3]
```diff
@@ -106,8 +111,13 @@ func main() {
 	// stopping workers
 	workers.Stop()
 
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
 	// wait until all workers will be stopped
-	workers.Wait()
+	if err = workers.Wait(ctx); err != nil {
+		fmt.Printf("[%s] error of completion\n", time.Now().Format("15:04:05"))
```
Comment (im-kulikov) [inline examples/workers/main.go:119]: srsly?

File: workers/pool.go [hunk 4]
```diff
@@ -1,7 +1,9 @@
 package workers
 
 import (
+	"context"
 	"sync"
+	stdAtomic "sync/atomic"
```
Comment (im-kulikov) [inline workers/pool.go:6]: why you use two atomics?

**Review by im-kulikov — CHANGES_REQUESTED** [review]
fmt to stdout? You're srsly?

File: examples/workers/main.go (hunk 3 above)
Comment (m1ome) [reply examples/workers/main.go:119]: This is examples. Looks normal for examples.

**Comment by m1ome** [comment]
@im-kulikov examples, it can be done in there.

**Comment by im-kulikov** [comment]
@m1ome we have logger... why not use it?

File: workers/pool.go (hunk 4 above)
Comment (jenchik) [reply workers/pool.go:6]: They are for different purposes

**Comment by jenchik** [comment]
This example shows only package of workers. The rest is optional

**Comment by im-kulikov** [comment]
In my opinion:
- artificial complication of the package 
- channel... channel.. channel

**Comment by im-kulikov** [comment]
simplest way https://github.com/robfig/cron/pull/99

**Comment by pcherednichenko** [comment]
it should not break tests
and so all ok

**Comment by im-kulikov** [comment]
duplicates #113

**Pull request closed without merging**